### PR TITLE
`SeamWebhook`: normalize headers before passing to svix + add docstrings

### LINKED ISSUE #84: Normalize headers to lowercase keys before passing to svix
HTTP headers are case-insensitive, however Svix may assume all headers have been normalized to lowercase. We know this is the case for Svix in C#.

The SeamWebhook class should do this normalization before passing the headers to Svix to protect against any upstream case-sensitive bugs.

See also https://github.com/seamapi/javascript-webhook/issues/7

### PR DESCRIPTION
**Pull request by andrii-balitskyi**
Closes https://github.com/seamapi/python/issues/84

### CODE REVIEW

File: seam/seam_webhook.py [hunk 1]
```diff
@@ -4,10 +4,40 @@
 
 
 class SeamWebhook:
+    """
+    A class to handle Seam webhook verification.
+
+    This class provides functionality to verify incoming webhook payloads
+    using the Svix library.
```
Comment (razor-x) [inline seam/seam_webhook.py]:
```suggestion
    Verifies and parses incoming Seam webhook events using the Svix library.
```

File: seam/seam_webhook.py [hunk 2]
```diff
@@ -4,10 +4,40 @@
...
+    """
+
     def __init__(self, secret: str):
+        """
+        Initialize the SeamWebhook instance.
+
+        Args:
+            secret (str): The secret key used for webhook verification.
```
Comment (razor-x) [inline seam/seam_webhook.py]: Where is this format taken from? If it's custom, let's instead adopt a standard format like [Sphinx](https://sphinx-rtd-tutorial.readthedocs.io/en/latest/docstrings.html#the-sphinx-docstring-format) that other tools are already able to parse.

Comment (andrii-balitskyi) [reply seam/seam_webhook.py]: Nice, I'll adopt the standard format.

Comment (andrii-balitskyi) [reply seam/seam_webhook.py]: https://github.com/seamapi/python/pull/113/commits/2782edde678e31c1b5e96edcd4ce88118a74465e

File: seam/__init__.py [hunk 3]
```diff
@@ -14,3 +14,4 @@
     SeamActionAttemptTimeoutError,
 )
 from .seam_webhook import SeamWebhook
+from svix.webhooks import WebhookVerificationError
```
Comment (andrii-balitskyi) [inline seam/__init__.py]: @razor-x, does it make sense to export the `WebhookVerificationError` raised by the `verify` method for SDK consumers to catch? Or should we catch it ourselves and raise a Seam-labeled error?

Comment (razor-x) [reply seam/__init__.py]: Ah, nice call. We should do this in https://github.com/seamapi/javascript-webhook too.

Can we alias it to `SeamWebhookVerificationError`? I'm not sure if we will be able to do this in all language, but it seems off to have `SeamWebhook` but then `WebhookVerificationError`.

File: seam/seam_webhook.py [hunk 4]
```diff
@@ -1,13 +1,33 @@
...
+        This method normalizes the headers, verifies the payload using the Svix
+        Webhook instance, and returns a SeamEvent object.
+
+        :param payload: The webhook event payload
+        :type payload: str
+        :param headers: A dictionary of HTTP headers
+        :type headers: Dict[str, str]
+        :return: An instance of SeamEvent created from the verified payload
+        :rtype: SeamEvent
+        :raises WebhookVerificationError: If the webhook signature verification fails
```
Comment (razor-x) [inline seam/seam_webhook.py]:
```suggestion
        :param payload: The raw HTTP request body.
        :type payload: str
        :param headers: The TTP request headers.
        :type headers: Dict[str, str]
        :return: The SeamEvent object created from the verified payload.
        :rtype: SeamEvent
        :raises WebhookVerificationError: If the webhook signature verification fails.
```

File: seam/seam_webhook.py [hunk 5]
```diff
@@ -1,13 +1,33 @@
...
+        :type secret: str
+        """
         self._webhook = Webhook(secret)
 
     def verify(self, payload: str, headers: Dict[str, str]) -> SeamEvent:
-        res = self._webhook.verify(payload, headers)
+        """Verify the incoming headers and webhook event payload.
+
+        This method normalizes the headers, verifies the payload using the Svix
+        Webhook instance, and returns a SeamEvent object.
```
Comment (razor-x) [inline seam/seam_webhook.py]:
```suggestion
        """Verify and parse an incoming HTTP webhook request.

        Normalizes the headers, verifies the payload using the Svix
        Webhook instance, and returns a SeamEvent object.
```

File: seam/seam_webhook.py [hunk 6]
```diff
@@ -1,13 +1,33 @@
...
+from svix.webhooks import Webhook
 from .routes.models import SeamEvent
 
 
 class SeamWebhook:
+    """Verifies and parses incoming Seam webhook events using the Svix library."""
+
     def __init__(self, secret: str):
+        """
+        :param secret: The secret key used for webhook verification
```
Comment (razor-x) [inline seam/seam_webhook.py]:
```suggestion
        :param secret: The secret key used for webhook verification.
```

**Review by razor-x — APPROVED** [review]

File: seam/seam_webhook.py (hunk 6 above)
Comment (andrii-balitskyi) [reply seam/seam_webhook.py]: Huge thank you for all the corrections!

File: seam/__init__.py (hunk 3 above)
Comment (andrii-balitskyi) [reply seam/__init__.py]: 16beac1006b82f84836043d44c60837fe6242f16

**Comment by DebbieAtSeam** [comment]
@andrii-balitskyi or @razor-x There are three items with added docstrings here. Is there a code sample that shows each of these three items in use (in context)? I'm asking because I'm not yet familiar with working with webhooks programmatically for Seam. Thanks!

**Comment by razor-x** [comment]
@DebbieAtSeam https://github.com/seamapi/python?tab=readme-ov-file#webhooks

**Pull request merged**